### DISCUSSION TITLE
CleanBlog - multiple blogs

### DISCUSSION THREAD

**Discussion by JohnGoldsmith** [statiqdev/Discussions#98] [Q&A]
Hi,  

I'm in the process of trying to convert my blog over to using Statiq and am using [CleanBlog](https://github.com/statiqdev/CleanBlog) as a base.  Is there a recommended approach for creating multiple blogs under the same scheme structure?

For example, where:

`PostSources: postsPlanes/* | postsTrains/* | postsAutomobiles/*`

For each of the three blogs there might be some shared assets and layouts, but alos some individual / per blog ones too.

At the moment I'm not quite sure how to approach carving this up - can anyone suggest a good approach?

Many thanks

John

**Comment by daveaglick** [comment]
This should actually be fairly straightforward - one of the big design goals in Statiq Web is to make everything driven by files and inputs (I.e. the generator itself doesn't make any assumptions about layout, file locations, etc.

In this case, the "blog" parts of CleanBlog are most driven by the `posts.cshtml` and `tags.cshtml` files. Both are [archive](https://www.statiq.dev/web/content-and-data/archives) files that aggregate multiple other documents (in this case, blog posts). The important thing to notice is the `ArchiveSources` front matter in each - that tells them where to find the files that will comprise the archive.

In CleanBlog, the `posts.cshtml` and `tags.cshtml` archives have their `ArchiveSources` set to a setting `PostSources`. That makes configuring the CleanBlog theme via a [settings](https://www.statiq.dev/framework/configuration/settings) file easier, but you can easily create additional archive files and/or override these ones with different values for `ArchiveSources` that doesn't look at a setting and can be set to whatever [globbing pattern](https://www.statiq.dev/framework/execution/files#globbing) you like.

Does that give you enough to go on?

**Comment by JohnGoldsmith** [comment]
That's great insight - thank you.  I hadn't spotted the front matter in the .cshtml file before and that makes more sense now.

I think I need to work through this for a while.  Right now I'm duplicating `index.cshtml` and `_layout.cshtml` so there are separate versions for each blog on the assumption that theses will have higher precedence than the theme ones and therefore override. [not sure if this is true]

Anyway, will post back here with my progress.

Thanks again for your help.
